nexusserver/server.go

```diff
@@ -0,0 +1,463 @@
+	if err != nil {
+		h.writeFailure(writer, err)
+	} else {
+		response.applyToStartResponse(writer, h)
+	}
+}
+
+func (h *httpHandler) getOperationResult(writer http.ResponseWriter, request *http.Request) {
+	// strip /result
+	prefix, operationID := path.Split(path.Dir(request.URL.Path))
+	operation := path.Base(prefix)
+	handlerRequest := &GetOperationResultRequest{Operation: operation, OperationID: operationID, HTTPRequest: request}
+
+	ctx := request.Context()
+	waitStr := request.URL.Query().Get(nexusapi.QueryWait)
+	if waitStr != "" {
+		waitDuration, err := time.ParseDuration(waitStr)
+		if err != nil {
+			h.logger.Warn("invalid wait duration query parameter", "wait", waitStr)
+			h.writeFailure(writer, newBadRequestError("invalid wait query parameter"))
+			return
+		}
+
+		var cancel func()
+		if waitDuration > h.options.GetResultMaxRequestTimeout {
+			waitDuration = h.options.GetResultMaxRequestTimeout
+		}
+		// TODO: reduce duration a bit to give some grace time?
+		ctx, cancel = context.WithTimeout(ctx, waitDuration)
+		handlerRequest.Wait = true
+		defer cancel()
+	}
+
+	response, err := h.options.Handler.GetOperationResult(ctx, handlerRequest)
+	if err != nil {
+		if handlerRequest.Wait && errors.Is(err, context.DeadlineExceeded) {
+			// Convert deadline exceeded to an async response to simplify handler implementation.
+			response = &OperationResponseAsync{OperationID: handlerRequest.OperationID}
+		} else {
+			h.writeFailure(writer, err)
+			return
+		}
+	}
+	response.applyToGetResultResponse(writer, h)
+}
+
+func (h *httpHandler) getOperationInfo(writer http.ResponseWriter, request *http.Request) {
+	prefix, operationID := path.Split(request.URL.Path)
+	operation := path.Base(prefix)
+	handlerRequest := &GetOperationInfoRequest{Operation: operation, OperationID: operationID, HTTPRequest: request}
+
+	info, err := h.options.Handler.GetOperationInfo(request.Context(), handlerRequest)
+	if err != nil {
+		h.writeFailure(writer, err)
+		return
+	}
+
+	bytes, err := h.options.Marshaler(info)
+	if err != nil {
+		h.writeFailure(writer, fmt.Errorf("failed to marshal operation info: %w", err))
+		return
+	}
+	writer.Header().Set(headerContentType, contentTypeJSON)
+	if _, err := writer.Write(bytes); err != nil {
+		h.logger.Error("failed to write response body", "error", err)
+	}
+}
+
+func (h *httpHandler) CancelOperation(writer http.ResponseWriter, request *http.Request) {
+	// strip /cancel
+	prefix, operationID := path.Split(path.Dir(request.URL.Path))
+	operation := path.Base(prefix)
+	handlerRequest := &CancelOperationRequest{Operation: operation, OperationID: operationID, HTTPRequest: request}
+
+	if err := h.options.Handler.CancelOperation(request.Context(), handlerRequest); err != nil {
+		h.writeFailure(writer, err)
+		return
+	}
+
+	writer.WriteHeader(http.StatusAccepted)
+}
+
+// Constructs an [http.Handler] from given options for handling Nexus service requests.
+func NewHTTPHandler(options Options) http.Handler {
```

Comment:
```suggestion
func New(options Options) *Server {
```

And have `*Server` implement `ServeHTTP` so it can be a handler. Returning interfaces is rough for future proofing.